README.md

```diff
@@ -271,28 +271,48 @@ function someFunction()
 
 <br/><br/>
 
-## ![✔] 3.4 Don't Forget the Semicolon
+## ![✔] 3.4 Separate your statements properly
 
-**TL;DR:** While not unanimously agreed upon, it is still recommended to put a semicolon at the end of each statement. This will make your code more readable and explicit to other developers who read it
+No matter if you use semicolons or not to separate your statements, knowing the common pitfalls of improper linebreaks or automatic semicolon insertion, will help you to eliminate regular syntax errors.
 
-**Otherwise:** As seen in the previous section, JavaScript's interpreter automatically adds a semicolon at the end of a statement if there isn't one, or considers a statement as not ended where it should, which might lead to some undesired results
+**TL;DR:** Use ESLint to gain awareness about separation concerns. [Prettier](https://prettier.io/) or [Standardjs](https://standardjs.com/) can automatically resolve these issues.
+
+**Otherwise:** As seen in the previous section, JavaScript's interpreter automatically adds a semicolon at the end of a statement if there isn't one, or considers a statement as not ended where it should, which might lead to some undesired results. You can use assignments and avoid using immediate invoked function expressions to prevent most of unexpected errors.
 
 ### Code example
 
 ```javascript
 // Do
-const count = 2;
-(function doSomething() {
-  // do something amazing
-}());
+function doThing() {
```

Comment:
@MartinMuzatko @BrunoScheufler 

Could we separate this into 2-3 very short and concise examples?

Reply:
I have the following suggestions, but would like to have your feedback:

**Do:** Define your used values in variables or function parameters
``` js
const items = [1, 2, 3]
items.map(item => item * 3)
```

``` js
function timesThree(numbers) {
    return numbers.map(number => number * 3)
}
```

**Avoid:** Unlogical order of statements - only define statements in an order as they are needed. If your code works in a side-effect free manner, then you would only define variables as late as they are needed.
``` js
const set = ['a', 'b', 'c']
[1, 2, 3].map(console.log)
// Uncaught TypeError: Cannot read property 'map' of undefined
```

Maybe you have suggestions for other examples?
As isaac states: https://blog.izs.me/2010/12/an-open-letter-to-javascript-leaders-regarding

>In general, \n ends a statement unless:
> 1. The statement has an unclosed paren, array literal, or object literal or ends in some other way that is not a valid way to end a statement. (For instance, ending with . or ,.)
> 2. The line is -- or ++ (in which case it will decrement/increment the next token.)
> 3. It is a for(), while(), do, if(), or else, and there is no {
> 4. The next line starts with [, (, +, *, /, -, ,, ., or some other binary operator that can only be found between two tokens in a single expression.

Reply:
Also I would maybe add:
> Clever short-hands are discouraged, in favor of clear and readable expressions, whenever possible.
> source: https://standardjs.com/rules.html#semicolons